### PR TITLE
refactor: replace hardcoded API URLs with dynamic endpoint URLs

### PR DESCRIPTION
**Pull request by psyray**
- Replaced hardcoded API URLs with dynamic endpoint URLs passed as function parameters.
- Updated various JavaScript functions to accept endpoint URLs as arguments.
- Modified HTML templates to include dynamic URLs in function calls.
- Added new modals for WAF Detector, CMS Detector, WHOIS Lookup, and CVE Details Lookup.
- Removed the getCurrentProjectSlug function as it is no longer needed.

## Summary by Sourcery

Refactor the codebase to replace hardcoded API URLs with dynamic endpoint URLs, enhancing flexibility and maintainability. Update JavaScript functions and HTML templates to support this change. Introduce new modals for various detectors and lookup functionalities, and remove obsolete functions.

New Features:
- Introduce new modals for WAF Detector, CMS Detector, WHOIS Lookup, and CVE Details Lookup to enhance user interaction and functionality.

Enhancements:
- Replace hardcoded API URLs with dynamic endpoint URLs passed as function parameters to improve flexibility and maintainability.
- Update JavaScript functions to accept endpoint URLs as arguments, allowing for more dynamic API interactions.
- Modify HTML templates to include dynamic URLs in function calls, ensuring consistency with the new dynamic URL approach.

Chores:
- Remove the getCurrentProjectSlug function as it is no longer needed due to the new dynamic URL handling.

### CODE REVIEW

**Comment by AnonymousWP** [comment]
Are you planning to solve these suggestions now or in a separate PR?

**Comment by psyray** [comment]
> Are you planning to solve these suggestions now or in a separate PR?

Yes when I will refactor all the JS.
But for now it's only bug fixes to make release 2.1.0 works well, no new feature or deep modifications

File: web/static/custom/toolbox.js [hunk 1]
```diff
@@ -1,62 +1,43 @@
...
-	`);
-	$('#modal_dialog').modal('show');
+	$('#whoisLookupModal').modal('show');
 }
 
-function toolbox_lookup_whois(){
+$(document).on('click', '#search_whois_toolbox_btn', function(){
 	var domain = document.getElementById("whois_domain_name").value;
+	var whoisLookupUrl = $(this).data('url');
+	var project_slug = $(this).data('slug');
```
[the remark this answers is not in the document]
Comment (AnonymousWP) [reply web/static/custom/toolbox.js:8]: See https://github.com/Security-Tools-Alliance/rengine-ng/pull/206#issuecomment-2377325716.

File: web/static/custom/toolbox.js [hunk 2]
```diff
@@ -1,62 +1,43 @@
...
-		</div>
-	`);
-	$('#modal_dialog').modal('show');
+	$('#whoisLookupModal').modal('show');
 }
 
-function toolbox_lookup_whois(){
+$(document).on('click', '#search_whois_toolbox_btn', function(){
 	var domain = document.getElementById("whois_domain_name").value;
+	var whoisLookupUrl = $(this).data('url');
```
[the remark this answers is not in the document]
Comment (AnonymousWP) [reply web/static/custom/toolbox.js:7]: See https://github.com/Security-Tools-Alliance/rengine-ng/pull/206#issuecomment-2377325716.

File: web/static/custom/todo.js [hunk 3]
```diff
@@ -141,8 +141,8 @@ function get_task_details(todo_id){
   });
 }
 
-function get_recon_notes(target_id, scan_id){
-  var url = `/api/listTodoNotes/?`;
+function get_recon_notes(endpoint, target_id, scan_id){
+  var url = `${endpoint}?`;
```
[the remark this answers is not in the document]
Comment (AnonymousWP) [reply web/static/custom/todo.js:145]: See https://github.com/Security-Tools-Alliance/rengine-ng/pull/206#issuecomment-2377325716.

**Review by AnonymousWP — APPROVED** [review]
Approving and merging because of https://github.com/Security-Tools-Alliance/rengine-ng/pull/206#issuecomment-2377325716.

**Pull request merged**